### DISCUSSION TITLE
How can I surface diagnostics from all buffers?

### DISCUSSION THREAD

**Discussion by andrewbanchich** [minad/consult#919] [Ideas]
I noticed that at some point I only see diagnostics from buffers which I open using `consult-flymake`. E.g. if I have errors in `foo.rs` and `bar.rs` and open `foo.rs`, `consult-flymake` only shows that one error. If I then open `bar.rs`, `consult-flymake` shows both errors.

Not sure if this is related, but even after that change, the diagnostic count in my modeline only shows the correct values for the open buffer.

Could you clarify how I can get the previous behavior back, where the diagnostics are for all files in my project, regardless of whether I've opened them?

I tried doing this (based on looking at [this PR](https://github.com/minad/consult/pull/589)), but it didn't have any effect. Elisp is not my strongpoint 😅 

```elisp

(defun consult-flymake-project ()
  "Invoke consult-flymake across all project buffers."
  (interactive)
  (consult-flymake 1)
  )

(use-package eglot
  :defer t
  :hook
  ((rust-mode rust-ts-mode typescript-mode typescript-ts-mode) . eglot-ensure)
  :bind
  (("C-d" . consult-flymake-project))

```

**Comment by minad** [comment]
Yes, giving a prefix argument to `consult-flymake` should yield all diagnostics, i.e., `C-u M-x consult-flymake`. I think you are hitting the problem #716, see also https://github.com/minad/consult/commit/5b2b7d39534314c53e95958c3d9daeee40329c28. A patch improving this would be welcome.

**Reply by andrewbanchich** [reply]
Weird, when I evaluate `(consult-flymake 1)` it doesn't log any errors. It just shows the diagnostics for any buffers which I've opened already.

Would love to help with a patch, but unfortunately I don't have the bandwidth to improve my elisp enough to contribute right anytime soon.

**Reply by minad** [reply]
> Weird, when I evaluate (consult-flymake 1) it doesn't log any errors. It just shows the diagnostics for any buffers which I've opened already.

I just gave it a quick test. I get a list of all flymake errors of all opened files in the current project.

**Reply by andrewbanchich** [reply]
Yeah, what I thought used to happen (or at least what I'd expect) is for it to provide diagnostics for all files in my project, unopened or not. If I change a type of a variable declared in one file then I need to know what broke throughout the rest of my files.

**Reply by minad** [reply]
> Yeah, what I thought used to happen (or at least what I'd expect) is for it to provide diagnostics for all files in my project, unopened or not. If I change a type of a variable declared in one file then I need to know what broke throughout the rest of my files.

Of course, this would be better, but is not yet supported. See #716.

**Reply by andrewbanchich** [reply]
Yep, just mentioning that that user is reporting they get an error, while I am not.

**Reply by minad** [reply]
Yes, back then then there was an error. Now we simply filter out diagnostics reported in non-opened files, see https://github.com/minad/consult/commit/5b2b7d39534314c53e95958c3d9daeee40329c28. We are going a bit in circles here, aren't we? ;)

**Reply by andrewbanchich** [reply]
Ah, I see. Thanks for clarifying!

**Comment by minad** [comment]
Moved to ideas to keep it in mind. It would be great if this gets implemented at some point.
